crates/core/src/store.rs

```diff
@@ -207,11 +208,19 @@ impl StoreBuilder {
             WasiCtxBuilder::Preview1(_) => {
                 panic!("Enabling network only allowed in preview2")
             }
-            WasiCtxBuilder::Preview2(ctx) => {
+            WasiCtxBuilder::Preview2(_) => {
                 // TODO: ctx.allow_udp(false);
-                ctx.inherit_network();
             }
         });
+
+        self.net_pool.insert_ip_net_port_any(
+            IpNet::V4(Ipv4Net::new(Ipv4Addr::new(0, 0, 0, 0), 0).unwrap()),
```

Comment:
I get the hesitation to do `"0.0.0.0/0".parse().unwrap()`, but I do prefer it for readability. Perhaps a comment stating that this adds all IP addresses will make this clear. It took me a second to realize this was a CIDR range and a comment would have helped.